docs/Lua API.rst

```diff
@@ -3617,8 +3627,27 @@ The class has the following methods:
 
   Calls ``onInput`` on all visible active subviews, iterating the ``subviews``
   sequence in *reverse order*, so that topmost subviews get events first.
-  Returns *true* if any of the subviews handled the event.
-
+  Returns ``true`` if any of the subviews handled the event. If a subview within
+  the view's ``focus_group`` has focus and it and all of its ancestors are
+  active and visible, that subview is offered the chance to handle the input
+  before any other subviews.
+
+* ``view:getPreferredFocusState()``
+
+  Returns ``false`` by default, but should be overridden by subclasses that may
+  want to take keyboard focus (if it is unclaimed) when they are added to a
+  parent view with ``addviews()``.
```

Comment:
So in an `A -> B -> C` hierarchy, if C requests focus, but A gave some child other than B focus, my understanding is that `C:setFocus(true)` would be called, `C.focus` would be set to true, but C would not actually receive keyboard events from A. Is that correct?

Reply:
Not quite. If C wants focus, but the focus group already has a focus owner, C does not get focus when it is added to the group.

If `C:setFocus(true)` is called by the controlling script after C is added to A's focus group, then the current focus owner loses focus and C gains focus.

If C already had focus when it was added to A's group, it would lose focus at that time.

Reply:
Since B and B2 are in the same hierarchy, they'll have the same focus group. This is ensured by the logic in `addviews()`.

The only way B and B2 could have different focus groups (and therefore two "focus owners") is if one of them hasn't been added to a Screen's hierarchy yet (the Screen would be the A node here), but then it would be getting keyboard events or being called for rendering anyway.